### ISSUE TITLE
PageBuilder - onSave callback on advanced element settings plugins is not executed

### ISSUE THREAD

**Issue by Pavel910** [webiny/webiny-js#2776]
### What are the steps to reproduce this bug?

Define a plugin with the `onSave` callback as depicted by the following plugin type:

```tsx
export type PbEditorPageElementAdvancedSettingsPlugin = Plugin & {
    type: "pb-editor-page-element-advanced-settings";
    elementType: string;
    render(params: { Bind: BindComponent; data: any; submit: () => void }): ReactElement;
    onSave?: (data: FormData) => FormData;
};
```

The `onSave` callback is not called when the settings form is submitted.

### What is the expected behavior?

The `onSave` callback is executed, and my `data` modifications are applied to the element.

### What do you see instead?

The callback is simply ignored, as if it's not there.

### Possible solution

Invoke the callback in this `onSubmit` handler here:
https://github.com/webiny/webiny-js/blob/dev/packages/app-page-builder/src/editor/plugins/elementSettings/advanced/ElementSettings.tsx#L15-L17

Also, make the callback `async` so developers can do asynchronous stuff here (like calling an external API).

**Comment by hemzet** [comment]
Hi Pavel, 
Does this ticket cover also what we're experiencing (see below)? Thanks

[..}
Uploaded image to page in page builder (edit ->settings). When refreshing, the image will disappear. And we also checked by API Playground, we found that the image was not stored in data of the page.
[..}

**Comment by Pavel910** [comment]
@hemzet no, this issue is not related. But, the issue you mentioned is resolved in #2804 :)
